frontend/public/components/dashboards-page/overview-dashboard/health-card.tsx

```diff
@@ -0,0 +1,176 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import * as plugins from '../../../plugins';
+import {
+  DashboardCard,
+  DashboardCardBody,
+  DashboardCardHeader,
+  DashboardCardTitle,
+  DashboardCardSeeAll,
+} from '../../dashboard/dashboard-card';
+import { HealthBody, HealthItem } from '../../dashboard/health-card';
+import { HealthState } from '../../dashboard/health-card/states';
+import { coFetch } from '../../../co-fetch';
+import { featureReducerName } from '../../../reducers/features';
+import { FLAGS } from '../../../const';
+import { withDashboardResources, DashboardItemProps } from '../with-dashboard-resources';
+import { RootState } from '../../../redux';
+import { getBrandingDetails } from '../../masthead';
+
+export const HEALTHY = 'is healthy';
+export const ERROR = 'is in an error state';
+
+const getClusterHealth = (subsystemStates: Array<SubsystemHealth>): ClusterHealth => {
+  let healthState: ClusterHealth = { state: HealthState.OK, message: 'Cluster is healthy' };
+  const subsystemBySeverity = {
+    error: subsystemStates.filter(subsystem => subsystem.state === HealthState.ERROR),
+    warning: subsystemStates.filter(subsystem => subsystem.state === HealthState.WARNING),
+    loading: subsystemStates.filter(subsystem => subsystem.state === HealthState.LOADING),
+  };
+
+  if (subsystemBySeverity.loading.length > 0) {
+    healthState = { state: HealthState.LOADING, message: null };
+  } else if (subsystemBySeverity.error.length > 0) {
+    healthState =
+      subsystemBySeverity.error.length === 1
+        ? subsystemBySeverity.error[0]
+        : { state: HealthState.ERROR, message: 'Multiple errors', details: 'Cluster health is degraded' };
+  } else if (subsystemBySeverity.warning.length > 0) {
+    healthState =
+      subsystemBySeverity.warning.length === 1
+        ? subsystemBySeverity.warning[0]
+        : { state: HealthState.WARNING, message: 'Multiple warnings', details: 'Cluster health is degraded' };
+  }
+
+  return healthState;
+};
+
+const getName = (isOpenShift: boolean): string => isOpenShift ? getBrandingDetails().productName : 'Kubernetes';
+
+const getK8sHealthState = (isOpenShift: boolean, k8sHealth: any): SubsystemHealth => {
+  if (!k8sHealth) {
+    return { state: HealthState.LOADING };
+  }
+  return k8sHealth === 'ok'
+    ? { message: `${getName(isOpenShift)} ${HEALTHY}`, state: HealthState.OK }
+    : { message: `${getName(isOpenShift)} ${ERROR}`, state: HealthState.ERROR };
+};
+
+const mapStateToProps = (state: RootState) => ({
+  isOpenShift: state[featureReducerName].get(FLAGS.OPENSHIFT),
```

Comment:
You're not checking if the flag is pending, so it could flash the wrong product name briefly on initial load.

Reply:
added check for pending flag. While the flag is pending we show loading in Health card body and See All link is not shown at all